pkg/cmd/server/kubernetes/master_config.go

```diff
@@ -147,14 +149,35 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
 		proxyClientCerts = append(proxyClientCerts, clientCert)
 	}
 
+	// TODO you have to know every APIGroup you're enabling or upstream will panic.  It's alternative to panicing is Fataling
+	// It needs a refactor to return errors
 	storageDestinations := master.NewStorageDestinations()
-	storageDestinations.AddAPIGroup("", databaseStorage)
+	// storageVersions is a map from API group to allowed versions that must include all the valid versions
+	// turns out, this must match values from storageDestinations or the master will fatal out, but
+	// the two are directly linked or validated any way upstream.  Upstream seems to be bugged and only allows
+	// a single version per group.
+	// TODO this will fail to compile once we fix upstream
+	storageVersions := map[string]string{}
+
+	enabledKubeVersions := configapi.GetEnabledAPIVersionsForGroup(*options.KubernetesMasterConfig, configapi.APIGroupKube)
+	enabledKubeVersionSet := sets.NewString(enabledKubeVersions...)
+	if len(enabledKubeVersions) > 0 {
+		storageDestinations.AddAPIGroup(configapi.APIGroupKube, databaseStorage)
+		storageVersions[configapi.APIGroupKube] = enabledKubeVersions[0]
```

Comment:
> Why is it okay to always use the first element?

Length is over zero, this arbitrarily chooses the first version until I can fix `storageVersions` upstream to handle multiple versions for one group.